packages/node_modules/@webex/internal-plugin-encryption/src/kms-certificate-validation.js

```diff
@@ -0,0 +1,229 @@
+import {parse as parseUrl} from 'url';
+
+import {isUri} from 'valid-url';
+import {fromBER} from 'asn1js';
+import {
+  Certificate,
+  RSAPublicKey,
+  CertificateChainValidationEngine,
+  CryptoEngine,
+  setEngine
+} from 'pkijs';
+import {isArray} from 'lodash';
+import jose from 'node-jose';
+import {Crypto} from '@peculiar/webcrypto';
+
+const webcrypto = new Crypto();
+
+setEngine(
+  'newEngine',
+  webcrypto,
+  new CryptoEngine({name: '', crypto: webcrypto, subtle: webcrypto.subtle})
+);
+
+const VALID_KTY = 'RSA';
+const VALID_KID_PROTOCOL = 'kms:';
+
+const X509_COMMON_NAME_KEY = '2.5.4.3';
+
+const X509_SUBJECT_ALT_NAME_KEY = '2.5.29.17';
+
+/**
+ * Customize Error so the SDK knows to quit retrying and notify
+ * the user
+ */
+export class KMSError extends Error {
+  /**
+   * add kmsError field to notify
+   * @param {string} message
+   */
+  constructor(message) {
+    super(message);
+    this.kmsError = true;
+  }
+}
+
+const throwError = (err) => {
+  throw new KMSError(`INVALID KMS: ${err}`);
+};
+
+/**
+ * Converts the PEM string to a pkijs certificate object
+ * @param {string} pem PEM representation of a certificate
+ * @returns {Certificate} pkijs object of the certificate
+ */
+const decodeCert = (pem) => {
+  if (typeof pem !== 'string') {
+    throwError('certificate needs to be a string');
+  }
+
+  const der = Buffer.from(pem, 'base64');
+  const ber = new Uint8Array(der).buffer;
+
+  const asn1 = fromBER(ber);
+
+  return new Certificate({schema: asn1.result});
+};
+
+/**
+ * Validate the 'kty' property of the KMS credentials
+ * @param {Object} JWT KMS credentials
+ * @param {string} JWT.kty type of certificate
+ * @throws if kty is not a valid type
```

Comment:
You could utilize your error object:
```suggestion
 * @throws {KMSError} if kty is not a valid type
```